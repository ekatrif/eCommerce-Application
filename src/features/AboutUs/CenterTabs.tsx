/* import * as React from 'react';
import Tabs from '@mui/material/Tabs';
import Tab from '@mui/material/Tab';
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';
import Avatar from '@mui/material/Avatar';

interface TabPanelProps {
  children?: React.ReactNode;
  index: number;
  value: number;
}

function TabPanel(props: TabPanelProps): JSX.Element {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`vertical-tabpanel-${index}`}
      aria-labelledby={`vertical-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box sx={{ p: 3 }}>
          <Typography>{children}</Typography>
        </Box>
      )}
    </div>
  );
}

interface Ia11yProps {
  id: string;
  'aria-controls': string;
}

function a11yProps(index: number): Ia11yProps {
  return {
    id: `vertical-tab-${index}`,
    'aria-controls': `vertical-tabpanel-${index}`,
  };
}

export default function VerticalTabs(): JSX.Element {
  const [value, setValue] = React.useState(0);

  const handleChange = (event: React.SyntheticEvent, newValue: number): void => {
    setValue(newValue);
  };

  return (
    <Box sx={{ flexGrow: 1, bgcolor: 'background.paper', display: 'flex', height: 224 }}>
      <Tabs
        orientation="vertical"
        variant="scrollable"
        value={value}
        onChange={handleChange}
        aria-label="Vertical tabs example"
        sx={{ borderRight: 1, borderColor: 'divider' }}
      >
        <Tab label={<Avatar alt="Remy Sharp" src="../../assets/images/team/oleg.jpg" />} {...a11yProps(0)} />
        <Tab label={<Avatar alt="Remy Sharp" src="../../assets/images/team/kate.jpg" />} {...a11yProps(1)} />
        <Tab label={<Avatar alt="Remy Sharp" src="../../assets/images/team/andrey.jpg" />} {...a11yProps(2)} />
      </Tabs>
      <TabPanel value={value} index={0}>
        Вклад Олега был в ...
      </TabPanel>
      <TabPanel value={value} index={1}>
        Вклад Кати был в ...
      </TabPanel>
      <TabPanel value={value} index={2}>
        Вклад Андрея был в ...
      </TabPanel>
    </Box>
  );
} */

import * as React from 'react';
import Tabs from '@mui/material/Tabs';
import Tab from '@mui/material/Tab';
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';
import Avatar from '@mui/material/Avatar';

interface TabPanelProps {
  children?: React.ReactNode;
  index: number;
  value: number;
}

function CustomTabPanel(props: TabPanelProps): JSX.Element {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`simple-tabpanel-${index}`}
      aria-labelledby={`simple-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box sx={{ p: 3 }}>
          <Typography>{children}</Typography>
        </Box>
      )}
    </div>
  );
}

interface Ia11yProps {
  id: string;
  'aria-controls': string;
}

function a11yProps(index: number): Ia11yProps {
  return {
    id: `simple-tab-${index}`,
    'aria-controls': `simple-tabpanel-${index}`,
  };
}

export default function CenterTabs(): JSX.Element {
  const [value, setValue] = React.useState(0);

  const handleChange = (event: React.SyntheticEvent, newValue: number): void => {
    setValue(newValue);
  };

  return (
    <Box sx={{ maxWidth: '850px' }}>
      <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
        <Tabs value={value} onChange={handleChange} aria-label="basic tabs example" centered>
          <Tab label={<Avatar alt="Remy Sharp" src="../../assets/images/team/oleg.jpg" />} {...a11yProps(0)} />
          <Tab label={<Avatar alt="Remy Sharp" src="../../assets/images/team/kate.jpg" />} {...a11yProps(1)} />
          <Tab label={<Avatar alt="Remy Sharp" src="../../assets/images/team/andrey.jpg" />} {...a11yProps(2)} />
        </Tabs>
      </Box>
      <CustomTabPanel value={value} index={0}>
        Вклад Олега был в ... Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita, qui autem! Rem atque at
        adipisci modi. Tempore vitae laborum cupiditate, quidem placeat obcaecati excepturi blanditiis molestiae
        dignissimos explicabo totam fuga. Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita, qui autem!
        Rem atque a adipisci modi. Tempore vitae laborum cupiditate, quidem placeat obcaecati excepturi blanditiis
        molestiae dignissimos explicabo totam fuga. quidem placeat obcaecati excepturi blanditiis molestiae dignissimos
        explicabo totam fuga. Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita, qui autem! Rem atque a
        adipisci modi. Tempore vitae laborum cupiditate, quidem placeat obcaecati excepturi blanditiis molestiae
        dignissimos explicabo.
      </CustomTabPanel>
      <CustomTabPanel value={value} index={1}>
        Вклад Кати был в ... Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita, qui autem! Rem atque at
        adipisci modi. Tempore vitae laborum cupiditate, quidem placeat obcaecati excepturi blanditiis molestiae
        dignissimos explicabo totam fuga. Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita, qui autem!
        Rem atque a adipisci modi. Tempore vitae laborum cupiditate, quidem placeat obcaecati excepturi blanditiis
        molestiae dignissimos explicabo totam fuga. quidem placeat obcaecati excepturi blanditiis molestiae dignissimos
        explicabo totam fuga. Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita, qui autem! Rem atque a
        adipisci modi. Tempore vitae laborum cupiditate, quidem placeat obcaecati excepturi blanditiis molestiae
        dignissimos explicabo.
      </CustomTabPanel>
      <CustomTabPanel value={value} index={2}>
        Вклад Андрея был в ... Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita, qui autem! Rem atque
        adipisci modi. Tempore vitae laborum cupiditate, quidem placeat obcaecati excepturi blanditiis molestiae
        dignissimos explicabo totam fuga. Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita, qui autem!
        Rem atque a adipisci modi. Tempore vitae laborum cupiditate, quidem placeat obcaecati excepturi blanditiis
        molestiae dignissimos explicabo totam fuga. quidem placeat obcaecati excepturi blanditiis molestiae dignissimos
        explicabo totam fuga. Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita, qui autem! Rem atque a
        adipisci modi. Tempore vitae laborum cupiditate, quidem placeat obcaecati excepturi blanditiis molestiae
        dignissimos explicabo.
      </CustomTabPanel>
    </Box>
  );
}
