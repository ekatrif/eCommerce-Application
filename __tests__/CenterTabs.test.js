import React from 'react';
import renderer from 'react-test-renderer';
import CenterTabs from '../src/features/AboutUs/CenterTabs';

test('CenterTabs renders correctly', () => {
  const tree = renderer
    .create(<CenterTabs />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
