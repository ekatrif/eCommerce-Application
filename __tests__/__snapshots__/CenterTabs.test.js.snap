// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterTabs renders correctly 1`] = `
<div
  className="MuiBox-root css-pzaa1o"
>
  <div
    className="MuiBox-root css-1gsv261"
  >
    <div
      className="MuiTabs-root css-1ujnqem-MuiTabs-root"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        style={
          {
            "marginBottom": -0,
            "overflow": "hidden",
          }
        }
      >
        <div
          aria-label="basic tabs example"
          aria-orientation={null}
          className="MuiTabs-flexContainer MuiTabs-centered css-1wf8b0h-MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-controls="simple-tabpanel-0"
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            disabled={false}
            id="simple-tab-0"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
            >
              <img
                alt="Remy Sharp"
                className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="../../assets/images/team/oleg.jpg"
              />
            </div>
            <span
              className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
              style={{}}
            />
          </button>
          <button
            aria-controls="simple-tabpanel-1"
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            disabled={false}
            id="simple-tab-1"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
            >
              <img
                alt="Remy Sharp"
                className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="../../assets/images/team/kate.jpg"
              />
            </div>
          </button>
          <button
            aria-controls="simple-tabpanel-2"
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            disabled={false}
            id="simple-tab-2"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
            >
              <img
                alt="Remy Sharp"
                className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="../../assets/images/team/andrey.jpg"
              />
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-labelledby="simple-tab-0"
    hidden={false}
    id="simple-tabpanel-0"
    role="tabpanel"
  >
    <div
      className="MuiBox-root css-19kzrtu"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Вклад Олега был в ... Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita, qui autem! Rem atque at adipisci modi. Tempore vitae laborum cupiditate, quidem placeat obcaecati excepturi blanditiis molestiae dignissimos explicabo totam fuga. Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita, qui autem! Rem atque a adipisci modi. Tempore vitae laborum cupiditate, quidem placeat obcaecati excepturi blanditiis molestiae dignissimos explicabo totam fuga. quidem placeat obcaecati excepturi blanditiis molestiae dignissimos explicabo totam fuga. Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita, qui autem! Rem atque a adipisci modi. Tempore vitae laborum cupiditate, quidem placeat obcaecati excepturi blanditiis molestiae dignissimos explicabo.
      </p>
    </div>
  </div>
  <div
    aria-labelledby="simple-tab-1"
    hidden={true}
    id="simple-tabpanel-1"
    role="tabpanel"
  />
  <div
    aria-labelledby="simple-tab-2"
    hidden={true}
    id="simple-tabpanel-2"
    role="tabpanel"
  />
</div>
`;
